project_quazar/lib/project_quazar/game/ship.ex

```diff
@@ -1,41 +1,66 @@
 defmodule Ship do
   @moduledoc """
   Contains all ship-related functionality.
-  
+
   Composed of:
   - Kinematics (Movable component: x/y position, velocity & angle),
   - Maximum & Current health,
   - Bullet damage
   """
-  
-  defstruct [:kinematics, :max_health, :health, :bullet_dmg]
-
-  @doc "Creates a new `Ship` at x,y. Health sets maximum and current health, and bullet_dmg sets damage"
-  def new_ship(px, py, angle, health, bullet_dmg) do
-    %__MODULE__{
-      kinematics: Movable.new_movable(px, py, 0, 0, angle),
-      max_health: health,
-      health: health,
-      bullet_dmg: bullet_dmg
-    }
+
+  defstruct [
+    :kinematics,
+    :max_health,
+    :health,
+    :type,
+    :size,
+    :acceleration,
+    :bullet_type
+  ]
+
+  # Bullet type specifications
+  # TODO: adjust size; bullet types
+  @ship_types %{
+    tank: %{health: 250, acceleration: 0.25, size: 6, bullet_type: :heavy},
+    destroyer: %{health: 150, acceleration: 0.5, size: 4, bullet_type: :medium},
+    scout: %{health: 100, acceleration: 1, size: 2, bullet_type: :light}
+  }
```

Comment:
Ship types are independent from selected bullet type, we should use the passed in value to create the bullet correctly.

Reply:
May need an extra validation function inside `Bullet` module to verify the passed in bullet_type string.

Reply:
Didn't we discuss that a ship can only fire one bullet type? And that bullet type is determined by ship type? 
I'll check again with Mike.

Reply:
Resolved, we are doing independent types. This will be handled in another PR. 👍